dropwizard/src/main/java/com/linecorp/armeria/dropwizard/connector/ArmeriaHttpConnectorFactory.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.dropwizard.connector;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.annotations.VisibleForTesting;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.server.ServerBuilder;
+
+import io.dropwizard.jetty.ConnectorFactory;
+import io.dropwizard.jetty.HttpConnectorFactory;
+import io.dropwizard.util.Size;
+import io.dropwizard.validation.MaxSize;
+import io.dropwizard.validation.MinSize;
+
+@JsonTypeName(ArmeriaHttpConnectorFactory.TYPE)
+public class ArmeriaHttpConnectorFactory extends HttpConnectorFactory
```

Comment:
I think we don't have to extend `HttpConnectorFactory`. The user who sets the properties in `HttpConnectorFactory` wouldn't know that the server is not working as expected right away.
So how about introducing a new class `ArmeriaConnectorFactory` and share the logic from `ArmeriaHttpConnectorFactory` and `ArmeriaHttpsConnectorFactory `? In that case, we can remove `ArmeriaServerDecorator`  as well. 😄

Reply:
Could. But then, I'd just be copying [large chunks of code out of `HttpConnectorFactory`](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-jetty/src/main/java/io/dropwizard/jetty/HttpConnectorFactory.java#L607-L700) and same for HttpsConnectorFactory

Reply:
I'm asking maybe I don't know much about how `Dropwizard` works 😄 but in the code that you gave to me, `HttpConnectorFactory` is making a new `ServerConnector` which has its own `acceptor`, `acceptChannel` and etc.
So are we accepting the connection using the `Connector`, not Armeria?

Reply:
I actually don't know that level of dropwizard myself.

If I did implement ConnectorFactory instead, how would I go about implementing the Connector class from armeria-jetty for the linked buildConnector method?

Reply:
I Don't know either. 😆 
I think we can use some classes in the jetty module and we can build our own connector factory so that Armeria server accepts all the connections. Just give me some time to look at the code in Jetty module and Dropwizard please. 😄

Reply:
> Could. But then, I'd just be copying large chunks of code out of HttpConnectorFactory  and same for HttpsConnectorFactory

So I found out that the code you linked which is  `HttpConnectorFactory.build()` is not called because we don't build the `connector` and add the `connector` to the `server` using `server.addConnector(connector)`.

But we have to support the configuration [properties](https://www.dropwizard.io/en/stable/manual/configuration.html) (even though we cannot support all of them), so I think we can just leave it as it is and add more properties that Armeria provides.

One thing I found is that if we add a fallback service, it doesn't work properly:
```java
public void configure(final ServerBuilder builder) {
    ....
    builder.serviceUnder("/", (ctx, res) -> HttpResponse.of("fall back!"));
}
```
It's because of [this](https://github.com/line/armeria/pull/2236/files#diff-2826258b9b33a09bf523f114e73b71faR280) so the requests which are matched by the router get into the `jettyService`. But I think we can handle it in the next PR because I think we might need to add and refactor our jetty module.

So, yes I think we are on the right way to go and I will do the rest of the review. Thanks and good job @cricket007!